### PR TITLE
Define all custom queries

### LINKED ISSUE #1: Finalize Rust interfaces
We have some decent interfaces for queries and messages, but there were some more requested by terra.

Add these other interfaces, and also add a `requires_terra` export.

### PR DESCRIPTION
**Pull request by ethanfrey**
Closes #1 

This defines the rest of the queries as described here: https://github.com/CosmWasm/cosmwasm/pull/211#issuecomment-613236209

I did some small refactoring and also add schema generation to `terra_bindings`, as well as a `requires_terra` flag

### CODE REVIEW

File: packages/bindings/examples/schema.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+use terra_bindings::{
+    ExchangeRateResponse, ExchangeRatesResponse, RewardsWeightResponse,
+    SeigniorageProceedsResponse, SimulateSwapResponse, TaxCapResponse, TaxProceedsResponse,
+    TaxRateResponse, TerraMsg, TerraQuery, TobinTaxResponse,
+};
+
+fn main() {
+    let mut out_dir = current_dir().unwrap();
+    out_dir.push("schema");
+    create_dir_all(&out_dir).unwrap();
```
Comment (webmaster128) [inline packages/bindings/examples/schema.rs:14]: The clean line is missing here (https://github.com/CosmWasm/cosmwasm/blob/v0.8.0-alpha3/contracts/queue/examples/schema.rs#L9-L12)

File: packages/bindings/src/lib.rs [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
-    TerraQuery,
+    ExchangeRateResponse, ExchangeRatesResponse, OracleQuery, RewardsWeightResponse,
+    SeigniorageProceedsResponse, SimulateSwapResponse, SwapQuery, TaxCapResponse,
+    TaxProceedsResponse, TaxRateResponse, TerraQuery, TobinTaxResponse, TreasuryQuery,
 };
+
+// This export is added to all contracts that import this package, signifying that they require
+// "terra" support on the chain they run on.
+#[no_mangle]
+extern "C" fn requires_terra() {}
```
Comment (webmaster128) [inline packages/bindings/src/lib.rs:16]: Do I understand correctly that the `terra` feature never exists in https://github.com/CosmWasm/cosmwasm but only contract+chain, passed intouched by the VM?

Comment (ethanfrey) [reply packages/bindings/src/lib.rs:16]: It never exists in cosmwasm, only for contracts that import these custom extensions.

File: packages/bindings/examples/schema.rs (hunk 1 above)
Comment (ethanfrey) [reply packages/bindings/examples/schema.rs:14]: good call

Comment (ethanfrey) [reply packages/bindings/examples/schema.rs:14]: good call

**Pull request merged**